gobblin-yarn/src/main/java/gobblin/yarn/YarnService.java

```diff
@@ -367,16 +441,23 @@ public void onContainersAllocated(List<Container> containers) {
       for (final Container container : containers) {
         LOGGER.info(String.format("Container %s has been allocated", container.getId()));
 
-        Map.Entry<Container, String> containerParticipantPair = new AbstractMap.SimpleImmutableEntry<>(
-            container, YarnHelixUtils.getParticipantId(container.getNodeId().getHost(), container.getId()));
-        containerMap.put(container.getId(), containerParticipantPair);
+        String instanceName = unusedHelixInstanceNames.poll();
+        if (Strings.isNullOrEmpty(instanceName)) {
+          // No unused instance name, so generating a new one.
+          instanceName = YarnHelixUtils.getHelixInstanceName(GobblinWorkUnitRunner.class.getSimpleName(),
+              helixInstanceIdGenerator.incrementAndGet());
+        }
+
+        final String finalInstanceName = instanceName;
+        containerMap.put(container.getId(), new AbstractMap.SimpleImmutableEntry<>(container, finalInstanceName));
```

Comment:
Can you prefix `containerMap` with `this`. This way when looking at a PR or RB it is clear this is an instance variable, not a local variable.